### PR TITLE
Guitar slide, hammeron and pulloff base class

### PR DESCRIPTION
**Pull request by laturetab**
Base class (Slurando) to track guitar slide, hammer-on and pull-offs. Save/loads information to/from mscx/mscz format. Draws slur between 2 notes like a tie, with optional text centered on slur or below staff (e.g. "P", "H", "Sl.") Currently does not have any UI and is not created by any importers, however it should work well with MusicXml and GuitarPro. Derived from tie.cpp and contains a lot of duplicate code. I will create a 2nd PR to remove the duplicate code if this class makes it into master.

- [x ] I signed [CLA](https://musescore.org/en/cla)
- [x ] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x ] I made sure the code compiles on my machine
- [x ] I made sure there are no unnecessary changes in the code
- [x ] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x ] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x ] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/libmscore/chord.cpp [hunk 1]
```diff
@@ -1986,6 +1987,15 @@ void Chord::layoutPitched()
             }
         }
 
+        // layout slurando
+        for (const Spanner* sp : note->spannerBack()) {
+            if (sp->isSlurando()) {
+                qreal d = ((Slurando*)sp)->layoutWidth(note, this);
```
Comment (Jojo-Schmitz) [inline src/libmscore/chord.cpp:1993]: Better have a `toSlurando()` here

Comment (laturetab) [reply src/libmscore/chord.cpp]: fixed

File: src/libmscore/chord.cpp [hunk 2]
```diff
@@ -2249,6 +2259,14 @@ void Chord::layoutTablature()
                 lll = qMax(lll, d);
             }
         }
+        // layout Slurando
+        for (const Spanner* sp : note->spannerBack()) {
+            if (sp->isSlurando()) {
+                qreal d = ((Slurando*)sp)->layoutWidthTab(note, this);
```
[the remark this answers is not in the document]
Comment (laturetab) [reply src/libmscore/chord.cpp]: fixed

**Comment by Jojo-Schmitz** [comment]
why?

**Pull request closed without merging**